tasks/webhook

```diff
@@ -62,12 +71,62 @@ def setup():
         shutil.copyfile(src, dest)
     os.umask(old_umask)
 
+def ensure_cockpit_checkout():
+    if not os.path.isdir(COCKPIT_CHECKOUT):
+        url = 'https://github.com/cockpit-project/cockpit/'
+        subprocess.check_call(['git', 'clone', url, COCKPIT_CHECKOUT])
+    subprocess.check_call(['git', '-C', COCKPIT_CHECKOUT, 'fetch', 'origin'])
+    subprocess.check_call(['git', '-C', COCKPIT_CHECKOUT, 'reset', '--hard'])
+    subprocess.check_call(['git', '-C', COCKPIT_CHECKOUT, 'clean', '-dxff'])
+    subprocess.check_call(['git', '-C', COCKPIT_CHECKOUT, 'checkout', 'origin/master'])
+
 
 class ReleaseHandler(github_handler.GithubHandler):
     def handle_event(self, event, request):
-        if event != 'create':
-            return (501, 'unsupported event ' + event)
+        if event == 'create':
+            return self.handle_create_event(event, request)
+        elif event == 'pull_request':
+            return self.handle_pull_request_event(event, request)
+        elif event == 'status':
+            return self.handle_status_event(event, request)
+        return (501, 'unsupported event ' + event)
+
+    def handle_pull_request_event(self, event, request):
+        if request['pull_request']['title'].endswith('[no-test]'):
+            return None
+        if request['action'] == 'closed':
+            return None
+
+        number = 0
+        try:
+            number = int(request['number'])
+        except (ValueError, TypeError):
+            return None
+
+        ensure_cockpit_checkout()
+        env = {}
```

Comment:
It's a bit surprising that `tests-scan` works with a completely empty environment. Shouldn't it at least maintain `$PATH`? Initializing to `os.environ.copy()` seems harmless enough.

Reply:
will do that

Reply:
actually I think I could just leave that out, since check_call with env=None would just inherit the current environment, no?

Reply:
Oh, you are right -- `GITHUB_BASE` would just get inherited.